lab-judy/route/gallery-router.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+//npm modules
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const debug = require('debug')('judy: gallery-route');
+const createError = require('http-errors');
+
+
+//app modules
+const bearerAuth = require('../lib/bearer-auth-middleware.js');
+const Gallery = require('../model/gallery.js');
+
+
+//constants
+const galleryRouter = module.exports = Router();
+
+galleryRouter.post('/api/gallery', bearerAuth, jsonParser, function(req, res, next){
+  debug('POST /api/gallery');
+
+  req.body.userID = req.user._id;
+  new Gallery(req.body).save()
+  .then( gallery => res.json(gallery))
+  .catch(next);
+});
+
+galleryRouter.get('/api/gallery/:id', bearerAuth, function(req, res, next){
+  debug('GET /api/gallery/:id');
+  Gallery.findById(req.params.id)
+  .then( gallery => {
+    if (gallery.userID.toString() !== req.user._id.toString())
+      return next(createError(401, 'invalid user ID'));
+    res.json(gallery);
+  })
+  .catch(next);
```

Comment:
This catch should pass a 404 error to next (created by createError) as if something went wrong that wasn't caught by the 401, it means that the ID was invalid and thus the gallery was 'Not Found.'